### PR TITLE
test(precompiles): add golden tests for policy registry V1 (BOP-426)

### PR DESCRIPTION
**Pull request by stephancill**
## What

Adds a golden/snapshot suite pinning **Policy Registry V1** behavior of the B-20 precompile (`crates/common/precompiles/tests/b20_policy_v1_golden.rs`), driven through the real `PolicyRegistryStorage` dispatch entry. Covers all 11 ops and the ticket edge cases:

- **createPolicy / createPolicyWithAccounts** — allowlist + blocklist, id encoding, `PolicyCreated`+`PolicyAdminUpdated` events, `ZeroAddress` and `BatchSizeTooLarge` guards.
- **Admin lifecycle** — `stageUpdateAdmin` (+clear), `finalizeUpdateAdmin`, `renounceAdmin`, with `Unauthorized` / `NoPendingAdmin` guards and `pendingPolicyAdmin`.
- **Membership** — `updateAllowlist` / `updateBlocklist` add **and** remove, with `IncompatiblePolicyType`, `Unauthorized`, `PolicyNotFound`, `BatchSizeTooLarge`.
- **Evaluation reads** — `isAuthorized` (empty allowlist rejects everyone, empty blocklist authorizes everyone, built-in `ALWAYS_ALLOW`/`ALWAYS_BLOCK`, malformed IDs), `policyExists`, `policyAdmin`, `pendingPolicyAdmin`.
- **Envelope** — `NonPayable`, and write-reverts-when-not-activated (reads bypass the activation gate).

Each case asserts exact returned bytes (or typed revert), resulting registry state, emitted events, and a per-case keccak storage-hash snapshot scoped to the registry address (independent of activation scaffolding); plus per-op gas footprints and a compile-time op-coverage checklist over `IPolicyRegistryCalls`.

## Why

Baseline for the frozen-manifest check (BOP-422/BOP-426). The pins were **authored and blessed against the shipped v1.1.1 policy-registry implementation**. Policy is not yet on the versioned seam — these goldens are the baseline that the future SOLID/versioned migration must satisfy unchanged (as already demonstrated for stablecoin, asset, and factory).

## Testing

```
cargo test -p base-common-precompiles --features test-utils --test b20_policy_v1_golden   # 26 passed
cargo test -p base-common-precompiles --features test-utils                               # full crate green
cargo clippy -p base-common-precompiles --features test-utils --all-targets               # clean
cargo +nightly fmt -p base-common-precompiles -- --check                                  # clean
```

Test-only: new integration test + one `Cargo.toml` `[[test]]` stanza; no source/logic changes.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>✅ Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
✅
<code>1/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by rayyan224 — DISMISSED** [review]

**Review by rayyan224 — APPROVED** [review]

File: crates/common/precompiles/tests/b20_policy_v1_golden.rs [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+fn golden_update_allowlist_remove() {
+    let mut s = fresh();
+    let id = create(&mut s, ADMIN, ADMIN, PolicyType::ALLOWLIST);
+    // Add then remove ALICE (exercises the membership-delete path).
+    call_policy(
+        &mut s,
+        ADMIN,
+        IPolicyRegistry::updateAllowlistCall { policyId: id, allowed: true, accounts: vec![ALICE] }
+            .abi_encode(),
+    );
```
[the remark this answers is not in the document]
Comment (stephancill) [reply crates/common/precompiles/tests/b20_policy_v1_golden.rs]: Good catch — bound and asserted success on both setup calls (`golden_update_allowlist_remove` add-ALICE, and `golden_finalize_reverts_unauthorized` stage) so a silent setup revert can no longer let the test pass for the wrong reason.

**Review by rayyan224 — APPROVED** [review]
